src/utils.js

```diff
@@ -310,4 +343,43 @@ export const convertSubfolderArray = (folderOrderArray, rawFolderContents, subfo
 export const generateImageorFilePath = (customPath, fileName) => {
   if (customPath) return encodeURIComponent(`${customPath}/${fileName}`)
   return fileName
+}
+
+const b64toBlob = (b64Data, contentType='', sliceSize=512) => {
+  const byteCharacters = atob(b64Data);
+  const byteArrays = [];
+
+  for (let offset = 0; offset < byteCharacters.length; offset += sliceSize) {
+    const slice = byteCharacters.slice(offset, offset + sliceSize);
+
+    const byteNumbers = new Array(slice.length);
+    for (let i = 0; i < slice.length; i++) {
+      byteNumbers[i] = slice.charCodeAt(i);
+    }
+
+    const byteArray = new Uint8Array(byteNumbers);
+    byteArrays.push(byteArray);
+  }
+
+  const blob = new Blob(byteArrays, {type: contentType});
+  return blob;
+}
+
+export async function fetchImageURL(siteName, filePath) {
+  var rawURL = `https://raw.githubusercontent.com/isomerpages/${siteName}/staging/${filePath}${filePath.endsWith('.svg') ? '?sanitize=true' : ''}`
+  const response = await fetch(rawURL);
```

Comment:
Attempting to retrieve the image here requires an extra call, which might not be very efficient - we could look into a one off check if the repo is public or private and run the blob generation code if the repo is private.

Reply:
Also for future reference, we mostly use axios, so we should try to standardise our request library as far as possible